astroscrappy/astroscrappy.pyx

```diff
@@ -183,14 +209,26 @@ def detect_cosmics(indat, inmask=None, float sigclip=4.5, float sigfrac=0.3,
         mask = np.empty((ny, nx), dtype=np.uint8, order='C')
         mask[:, :] = inmask[:, :]
 
-    # Add back in the previously subtracted sky level and multiply by the gain
-    # The statistics only work properly with electrons.
-    cleanarr += pssl
+    # Multiply by the gain; the statistics only work properly with electrons.
     cleanarr *= gain
```

Comment:
Maybe it would be clearer to also multiply `bkg` by `gain` and `var` by `gain ** 2.` up here, so you don't have to do it in the new lines below.

Reply:
Indeed, this would make the code simpler and improve the performance. I think there was a reason not to do this, but probably on an older version of this branch.

Reply:
There was a reason for this, but I've forgotten what it was and therefore will do this.

Reply:
I think this was because bkg was passed in by the user so I didn't want to change the variable while running the code. But this is only an issue in cases when the user is doing strange things with parallel threads so probably worth changing.